scalgoprotoc/rust_generator.py

```diff
@@ -535,27 +558,49 @@ def generate_struct_out(self, node: Value, lname: str) -> None:
             )
 
     def generate_table_in(self, node: Value, lname: str) -> None:
+        assert node.table is not None
         tname = node.table.name
-        if not node.inplace:
-            self.output_doc(node, "    ")
-            self.o(
-                f"""    #[inline]
-    pub fn {lname}(&self) -> Result<std::option::Option<{tname}In>> {{
-        self._reader.get_table::<{tname}In>({node.offset})
-    }}
-"""
-            )
+        if node.optional:
+            if not node.inplace:
+                self.output_doc(node, "    ")
+                self.o(
+                    f"""    #[inline]
+        pub fn {lname}(&self) -> Result<std::option::Option<{tname}In>> {{
+            self._reader.get_optional_table::<{tname}In>({node.offset})
+        }}
+    """
+                )
+            else:
+                self.output_doc(node, "    ")
+                self.o(
+                    f"""    #[inline]
+        pub fn {lname}(&self) -> Result<std::option::Option<{tname}In>> {{
+            self._reader.get_optional_table_inplace::<{tname}In>({node.offset})
+        }}
+    """
+                )
         else:
-            self.output_doc(node, "    ")
-            self.o(
-                f"""    #[inline]
-    pub fn {lname}(&self) -> Result<std::option::Option<{tname}In>> {{
-        self._reader.get_table_inplace::<{tname}In>({node.offset})
-    }}
-"""
-            )
+            if not node.inplace:
+                self.output_doc(node, "    ")
+                self.o(
+                    f"""    #[inline]
+        pub fn {lname}(&self) -> Result<{tname}In> {{
+            self._reader.get_table::<{tname}In>({node.offset})
+        }}
+    """
```

Comment:
Remove extra whitespace at the end of this code, and remove the extra indentation of the Rust code